tests/helpers/app-fixture.js

```diff
@@ -0,0 +1,253 @@
+'use strict';
+
+var fs = require('fs-extra');
+var path = require('path');
+var symlinkOrCopySync = require('symlink-or-copy').sync;
+var merge = require('ember-cli-lodash-subset').merge;
+
+var fixturify = require('fixturify');
+var quickTemp = require('quick-temp');
+
+var originalWorkingDirectory = process.cwd();
+var root = path.resolve(__dirname, '..', '..');
+
+var PackageCache = require('../../tests/helpers/package-cache');
+var CommandGenerator = require('../../tests/helpers/command-generator');
+
+var ember = new CommandGenerator(path.join(root, 'bin', 'ember'));
+
+function AppFixture(name) {
+  this.type = 'app';
+  this.command = 'new';
+  this.name = name;
+  this._installedAddons = [];
+
+  this._init();
+}
+
+AppFixture.prototype = {
+
+  _init: function() {
+    process.chdir(root);
+    this.dir = quickTemp.makeOrRemake({}, this.name + '-' + this.type + '-fixture');
+    process.chdir(originalWorkingDirectory);
+
+    this._loadBlueprint();
+  },
+
+  _loadBlueprint: function() {
+    fs.removeSync(this.dir);
+
+    ember.invoke(
+      this.command,
+      this.name,
+      '--directory=' + this.dir,
+      '--disable-analytics',
+      '--watcher=node',
+      '--skip-npm',
+      '--skip-bower',
+      '--skip-git'
+    );
+
+    this.fixture = fixturify.readSync(this.dir);
+
+    // Clean up after the generator.
+    fs.removeSync(this.dir);
+  },
+
+  serialize: function(isChild) {
```

Comment:
This is extended by `addon-fixture` which is extended by `in-repo-addon-fixture`. If any of these items has a `parent` it changes the way in which it does serialization.